src/Drupal/Commands/core/WatchdogCommands.php

```diff
@@ -62,9 +64,30 @@ public function show($substring = '', $options = ['format' => 'table', 'count' =
             $row = $this->formatResult($result, $options['extended']);
             $table[$row->wid] = (array)$row;
         }
-        if (empty($table)) {
+        if (empty($table) && !$tail) {
             $this->logger()->notice(dt('No log messages available.'));
         } else {
+            if ($tail) {
+                // Query for new watchdog messages every 2 seconds.
+                while (true) {
+                    $rsc = Database::getConnection()->select('watchdog', 'w')
+                        ->fields('w')
+                        ->condition('wid', $last_wid, '>')
+                        ->orderBy('wid', 'ASC')
+                        ->execute();
+                    while ($result = $rsc->fetchObject()) {
```

Comment:
Won't we be selecting all records since the beginning of time in the table?  That could be destructive.  I think we'd want to find the max WID in the table, and set that to the initial $last_wid.  Also, $last_wid isn't defined yet when we use it on the first trip through.

Reply:
yes; not defined; but `ws` is all ready scoping it to 10 records.